web/src/components/TaxRegistrationModal.tsx

```diff
@@ -0,0 +1,113 @@
+import { Content } from "@/components/Content";
+import { ModalTwoButton } from "@/components/ModalTwoButton";
+import { ProfileDataContext } from "@/contexts/profileDataContext";
+import { useConfig } from "@/lib/data-hooks/useConfig";
+import { useUserData } from "@/lib/data-hooks/useUserData";
+import { createProfileFieldErrorMap, ProfileFieldErrorMap, ProfileFields } from "@/lib/types/types";
+import { createEmptyProfileData, ProfileData, TaskProgress, UserData } from "@businessnjgovnavigator/shared";
+import { ReactElement, useEffect, useState } from "react";
+import { OnboardingBusinessName } from "./onboarding/OnboardingBusinessName";
+import { OnboardingExistingEmployees } from "./onboarding/OnboardingExistingEmployees";
+import { OnboardingOwnership } from "./onboarding/OnboardingOwnership";
+import { OnboardingTaxId } from "./onboarding/OnboardingTaxId";
+
+interface Props {
+  isOpen: boolean;
+  close: () => void;
+  onSave: (
+    newValue: TaskProgress,
+    userData: UserData,
+    { redirectOnSuccess }: { redirectOnSuccess: boolean }
+  ) => void;
+}
+
+export const TaxRegistrationModal = (props: Props): ReactElement => {
+  const { Config } = useConfig();
+  const { userData, update } = useUserData();
+  const [profileData, setProfileData] = useState<ProfileData>(createEmptyProfileData());
+  const [fieldStates, setFieldStates] = useState<ProfileFieldErrorMap>(createProfileFieldErrorMap());
+
+  useEffect(() => {
+    if (!userData) return;
+    setProfileData(userData.profileData);
+  }, [userData]);
+
+  const onValidation = (field: ProfileFields, invalid: boolean) => {
+    setFieldStates({ ...fieldStates, [field]: { invalid } });
+  };
+
+  const onSubmit = async () => {
+    if (!userData) return;
+    const errorMap = {
+      ...fieldStates,
+      businessName: { invalid: !profileData.businessName },
+      existingEmployees: { invalid: !profileData.existingEmployees },
+      OnboardingTaxId: { invalid: !profileData.taxId },
+    };
+    setFieldStates(errorMap);
+    if (Object.keys(errorMap).some((k) => errorMap[k as ProfileFields].invalid)) return;
+
+    const updatedUserData = { ...userData, profileData };
+    await update(updatedUserData);
+    props.onSave("COMPLETED", updatedUserData, { redirectOnSuccess: true });
+    props.close();
+  };
+
+  const showBusinessField = () => {
+    return (
+      userData?.profileData.businessName === "" &&
+      userData?.profileData.legalStructureId !== "sole-proprietorship" &&
+      userData?.profileData.legalStructureId !== "general-partnership"
+    );
+  };
+
+  const showTaxIdField = () => {
+    return (
+      userData?.profileData.legalStructureId !== "sole-proprietorship" &&
+      userData?.profileData.legalStructureId !== "general-partnership"
```

Comment:
for this and the function above, we should be using our `LookupByLegalStructure` function because it already tracks which legal structure IDs are tradeName versus publicFiling. We shouldn't have the ids hardcoded here